### ISSUE TITLE
Multiple gpus (1080Ti) do not speed up training, test on cifar10_estimator code

### ISSUE THREAD

**Issue by zihuaweng** [tensorflow/tensorflow#23032]
I was trying to test the performance of multi GPUs version of [cifar10_estimator][1] on 2 or 3 1080Ti, but received no speed-up.

I find some useful information about hardware [here][2], but still confused how to solve it.

My environment:
 - docker image tensorflow/tensorflow:latest-gpu-py3
 - Ubuntu VERSION=16.04.5 LTS
 - Python3
 - CUDA_VERSION=9.0.176
 - tensorflow-gpu=1.11.0

GPU info:

    nvidia-smi topo -m
    
    	GPU0	GPU1	GPU2	GPU3	GPU4	GPU5	GPU6	GPU7	CPU Affinity
    GPU0	 X 	PIX	PHB	PHB	SYS	SYS	SYS	SYS	0-7
    GPU1	PIX	 X 	PHB	PHB	SYS	SYS	SYS	SYS	0-7
    GPU2	PHB	PHB	 X 	PIX	SYS	SYS	SYS	SYS	0-7
    GPU3	PHB	PHB	PIX	 X 	SYS	SYS	SYS	SYS	0-7
    GPU4	SYS	SYS	SYS	SYS	 X 	PIX	PHB	PHB	8-15
    GPU5	SYS	SYS	SYS	SYS	PIX	 X 	PHB	PHB	8-15
    GPU6	SYS	SYS	SYS	SYS	PHB	PHB	 X 	PIX	8-15
    GPU7	SYS	SYS	SYS	SYS	PHB	PHB	PIX	 X 	8-15

    Legend:
    
      X    = Self
      SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
      NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
      PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
      PXB  = Connection traversing multiple PCIe switches (without traversing the PCIe Host Bridge)
      PIX  = Connection traversing a single PCIe switch
      NV#  = Connection traversing a bonded set of # NVLinks

1 gpu bach_size=128

    INFO:tensorflow:loss = 2.2576141, step = 200 (3.729 sec)
    INFO:tensorflow:learning_rate = 0.1, loss = 2.2576141 (3.729 sec)
    INFO:tensorflow:Average examples/sec: 2821.06 (2858.65), step = 200
    INFO:tensorflow:Average examples/sec: 2847.23 (3496.06), step = 210
    INFO:tensorflow:Average examples/sec: 2857.91 (3102.29), step = 220
    INFO:tensorflow:Average examples/sec: 2867.04 (3083.62), step = 230
    INFO:tensorflow:Average examples/sec: 2889.21 (3514.15), step = 240
    INFO:tensorflow:Average examples/sec: 2913.15 (3636.28), step = 250
    INFO:tensorflow:Average examples/sec: 2915.99 (2988.94), step = 260
    INFO:tensorflow:Average examples/sec: 2901.94 (2578.95), step = 270
    INFO:tensorflow:Average examples/sec: 2888.87 (2575.46), step = 280
    INFO:tensorflow:Average examples/sec: 2892.13 (2986.66), step = 290
    INFO:tensorflow:global_step/sec: 24.25

2 gpu bach_size=256 

    INFO:tensorflow:loss = 2.4630964, step = 200 (5.971 sec)
    INFO:tensorflow:learning_rate = 0.1, loss = 2.4630964 (5.971 sec)
    INFO:tensorflow:Average examples/sec: 3255.68 (4296.71), step = 200
    INFO:tensorflow:Average examples/sec: 3297.51 (4437.93), step = 210
    INFO:tensorflow:Average examples/sec: 3332.15 (4275.33), step = 220
    INFO:tensorflow:Average examples/sec: 3363.86 (4254.65), step = 230
    INFO:tensorflow:Average examples/sec: 3395.09 (4316.94), step = 240
    INFO:tensorflow:Average examples/sec: 3418.44 (4094.23), step = 250
    INFO:tensorflow:Average examples/sec: 3447.17 (4364.24), step = 260
    INFO:tensorflow:Average examples/sec: 3474.56 (4379.02), step = 270
    INFO:tensorflow:Average examples/sec: 3492.73 (4067.13), step = 280
    INFO:tensorflow:Average examples/sec: 3514.19 (4244.23), step = 290
    INFO:tensorflow:global_step/sec: 16.6026

3 gpu bach_size=384

    INFO:tensorflow:loss = 2.0980535, step = 200 (9.329 sec)
    INFO:tensorflow:learning_rate = 0.1, loss = 2.0980535 (9.329 sec)
    INFO:tensorflow:Average examples/sec: 3214.65 (4165.7), step = 200
    INFO:tensorflow:Average examples/sec: 3272.85 (5130.99), step = 210
    INFO:tensorflow:Average examples/sec: 3324.15 (4955.13), step = 220
    INFO:tensorflow:Average examples/sec: 3376.65 (5174.76), step = 230
    INFO:tensorflow:Average examples/sec: 3425.48 (5132.15), step = 240
    INFO:tensorflow:Average examples/sec: 3468.29 (4954.35), step = 250
    INFO:tensorflow:Average examples/sec: 3509.91 (5014.23), step = 260
    INFO:tensorflow:Average examples/sec: 3544.29 (4755.56), step = 270
    INFO:tensorflow:Average examples/sec: 3579.69 (4901.39), step = 280
    INFO:tensorflow:Average examples/sec: 3617.84 (5156.66), step = 290
    INFO:tensorflow:global_step/sec: 13.1009

[![enter image description here][3]][3]

  [1]: https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10_estimator
  [2]: https://github.com/rossumai/keras-multi-gpu/blob/master/blog/docs/hardware.md
  [3]: https://i.stack.imgur.com/yqIZX.png

**Comment by ppwwyyxx** [comment]
Many of the Tensorflow's official examples are not optimally written for performance. I think what you saw is quite normal. You can check out https://github.com/tensorflow/benchmarks/ which is written for performance.

**Comment by zihuaweng** [comment]
@ppwwyyxx Thanks!
I try [tf_cnn_benchmarks](https://github.com/tensorflow/benchmarks/tree/master/scripts/tf_cnn_benchmarks) using synthetic data and docker image tensorflow/tensorflow:nightly-gpu-py3, it works pretty well!

 1 gpu batch_size=32
```
Step	Img/sec	total_loss
1	images/sec: 198.0 +/- 0.0 (jitter = 0.0)	8.458
10	images/sec: 198.1 +/- 0.2 (jitter = 0.3)	7.997
20	images/sec: 197.9 +/- 0.3 (jitter = 0.6)	8.259
30	images/sec: 198.1 +/- 0.2 (jitter = 0.5)	8.340
40	images/sec: 198.0 +/- 0.2 (jitter = 0.6)	8.188
50	images/sec: 198.1 +/- 0.2 (jitter = 0.6)	7.754
60	images/sec: 198.2 +/- 0.1 (jitter = 0.6)	8.080
70	images/sec: 198.1 +/- 0.1 (jitter = 0.6)	8.472
80	images/sec: 198.2 +/- 0.1 (jitter = 0.6)	8.290
90	images/sec: 198.2 +/- 0.1 (jitter = 0.6)	8.041
100	images/sec: 198.1 +/- 0.1 (jitter = 0.6)	7.993
----------------------------------------------------------------
total images/sec: 197.98
```
2 gpus batch_size=32
```
Step	Img/sec	total_loss
1	images/sec: 373.2 +/- 0.0 (jitter = 0.0)	8.453
10	images/sec: 375.8 +/- 0.4 (jitter = 0.4)	8.209
20	images/sec: 375.5 +/- 0.5 (jitter = 2.5)	8.120
30	images/sec: 375.4 +/- 0.4 (jitter = 2.4)	8.235
40	images/sec: 375.6 +/- 0.4 (jitter = 2.3)	8.108
50	images/sec: 375.9 +/- 0.4 (jitter = 2.4)	8.027
60	images/sec: 375.8 +/- 0.3 (jitter = 2.6)	8.239
70	images/sec: 375.7 +/- 0.3 (jitter = 2.4)	8.373
80	images/sec: 375.4 +/- 0.3 (jitter = 2.5)	8.191
90	images/sec: 375.5 +/- 0.3 (jitter = 2.5)	7.976
100	images/sec: 375.6 +/- 0.3 (jitter = 2.8)	8.157
----------------------------------------------------------------
total images/sec: 375.40
```
3 gpus batch_size=32
```
Step	Img/sec	total_loss
1	images/sec: 524.8 +/- 0.0 (jitter = 0.0)	8.394
10	images/sec: 525.4 +/- 3.4 (jitter = 12.1)	8.143
20	images/sec: 527.6 +/- 2.4 (jitter = 10.1)	8.066
30	images/sec: 529.1 +/- 1.9 (jitter = 9.2)	8.191
40	images/sec: 529.0 +/- 1.5 (jitter = 8.9)	8.230
50	images/sec: 530.1 +/- 1.4 (jitter = 9.2)	8.049
60	images/sec: 530.5 +/- 1.2 (jitter = 9.2)	8.292
70	images/sec: 529.9 +/- 1.2 (jitter = 9.5)	8.247
80	images/sec: 530.4 +/- 1.1 (jitter = 9.2)	8.152
90	images/sec: 530.6 +/- 1.1 (jitter = 9.1)	8.019
100	images/sec: 530.5 +/- 1.0 (jitter = 8.6)	8.212
----------------------------------------------------------------
total images/sec: 530.16
```

**Comment by Latootoo** [comment]
> synthetic data

 I have tested this benchmark on ImageNet-2012. The training speed is similar to your previous work. However, It's perfect on the synthetic data.
